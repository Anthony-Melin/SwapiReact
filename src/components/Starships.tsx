import { PropsWithChildren } from 'react'

type StarshipsProps = PropsWithChildren<{ values: string[] }>

const Starships = ({ values, ...restProps }: StarshipsProps) => {
    return (
        <div data-testid="starships" {...restProps}>
            <p data-testid="starships-title">Starships</p>
            {values.length === 0 ? (
                <p data-testid="starships-empty">No starships linked</p>
            ) : (
                <ul>
                    {values.map((value, index) => {
                        return (
                            <li
                                key={`starships-${index}`}
                                data-testid={`starships-${index}`}
                            >
                                {value}
                            </li>
                        )
                    })}
                </ul>
            )}
        </div>
    )
}

export default Starships
