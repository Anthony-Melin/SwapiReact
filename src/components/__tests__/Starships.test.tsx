import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import Starships from '../Starships.tsx'

describe('Starships', () => {
    describe('with empty list', () => {
        beforeEach(() => {
            render(<Starships values={[]} />)
        })

        it('should match snapshot', () => {
            expect(screen.getByTestId('starships')).toMatchSnapshot()
        })

        it('should have a title', () => {
            expect(screen.getByTestId('starships')).toHaveTextContent(
                'Starships'
            )
        })

        it('should have empty list label', () => {
            expect(screen.getByTestId('starships-empty')).toHaveTextContent(
                'No starships linked'
            )
        })
    })

    describe('with filled list', () => {
        beforeEach(() => {
            render(<Starships values={['VALUE_1', 'VALUE_2']} />)
        })

        it('should match snapshot', () => {
            expect(screen.getByTestId('starships')).toMatchSnapshot()
        })

        it('should have a title', () => {
            expect(screen.getByTestId('starships')).toHaveTextContent(
                'Starships'
            )
        })

        it('should contains elements', () => {
            expect(screen.getByTestId('starships-0')).toHaveTextContent(
                'VALUE_1'
            )
            expect(screen.getByTestId('starships-1')).toHaveTextContent(
                'VALUE_2'
            )
        })
    })
})
