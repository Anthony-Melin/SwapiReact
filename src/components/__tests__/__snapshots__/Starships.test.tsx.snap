// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Starships > with empty list > should match snapshot 1`] = `
<div
  data-testid="starships"
>
  <p
    data-testid="starships-title"
  >
    Starships
  </p>
  <p
    data-testid="starships-empty"
  >
    No starships linked
  </p>
</div>
`;

exports[`Starships > with filled list > should match snapshot 1`] = `
<div
  data-testid="starships"
>
  <p
    data-testid="starships-title"
  >
    Starships
  </p>
  <ul>
    <li
      data-testid="starships-0"
    >
      VALUE_1
    </li>
    <li
      data-testid="starships-1"
    >
      VALUE_2
    </li>
  </ul>
</div>
`;
